Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialGraphCompiler.cpp

```diff
@@ -1176,8 +1179,7 @@ namespace MaterialCanvas
             for (const auto& [materialPropertyValueSlot, materialPropertyValueSlotConfig] : materialPropertyValueSlots)
             {
                 // Sampler states are currently not configurable and will not be added added to the material type, just the material SRG.
-                if (!materialPropertyValueSlot || materialPropertyValueSlot->GetValue().empty() ||
-                    materialPropertyValueSlot->GetValue().is<AZ::RHI::SamplerState>())
+                if (!materialPropertyValueSlot || materialPropertyValueSlot->GetValue().empty())
                 {
```

Comment:
The comments need to be updated but this should be fantastic. Have you tried creating a material type and changing sampler states from the material editor or material component? This should allow for dynamic sampler state changes which was not supported before.